packages/web-components/src/components/ic-chip/ic-chip.tsx

```diff
@@ -0,0 +1,224 @@
+    this.isFocussed = true;
+  };
+
+  private parentBlurred = (): void => {
+    this.isFocussed = false;
+  };
+
+  componentWillLoad(): void {
+    this.parentEl = this.el.parentElement;
+
+    if (this.parentEl.tagName === "A") {
+      this.clickable = true;
+      this.parentIsAnchorTag = true;
+      this.parentEl.classList.add("ic-chip-wrapper-link");
+      this.parentEl.addEventListener("focus", this.parentFocussed);
+      this.parentEl.addEventListener("blur", this.parentBlurred);
+    }
+
+    if (!this.dismissible) {
+      this.clickable = false;
+    }
+  }
+
+  disconnectedCallback(): void {
+    if (this.parentIsAnchorTag) {
+      this.parentEl.removeEventListener("focus", this.parentFocussed);
+      this.parentEl.removeEventListener("blur", this.parentBlurred);
+    }
+  }
+
+  componentDidLoad(): void {
+    onComponentRequiredPropUndefined(
+      [{ prop: this.label, propName: "label" }],
+      "Chip"
+    );
+  }
+
+  render() {
+    const {
+      label,
+      appearance,
+      size,
+      dismissible,
+      visible,
+      clickable,
+      disabled,
+      parentIsAnchorTag,
+      isFocussed,
+      href,
+      hreflang,
+      referrerpolicy,
+      rel,
+      target,
+    } = this;
+
+    const Component = parentIsAnchorTag
+      ? "div"
+      : clickable
+      ? this.href === undefined
+        ? "button"
+        : "a"
+      : "div";
+
+    const attrs = Component == "a" && {
+      href: href,
+      hrefLang: hreflang,
+      referrerPolicy: referrerpolicy,
+      rel: rel,
+      target: target,
+    };
+
+    return (
+      visible && (
+        <Component
+          class={{
+            ["chip"]: true,
+            [`${appearance}`]: true,
+            [`${size}`]: true,
+            ["clickable"]: clickable && !disabled,
+            ["disabled"]: disabled,
+            ["focussed"]: isFocussed,
+            ["dismissible"]: dismissible,
+          }}
+          tabindex={clickable && !parentIsAnchorTag ? 0 : null}
+          aria-disabled={disabled ? "true" : null}
+          disabled={disabled ? true : null}
+          {...attrs}
+        >
+          <div class="container">
+            {isSlotUsed(this.el, "icon") && (
+              <div class="icon">
+                <slot name="icon" />
+              </div>
+            )}
+            <ic-typography
+              variant="label"
+              apply-vertical-margins={false}
+              class="label"
+            >
+              <span>{label}</span>
+            </ic-typography>
+            {dismissible && (
+              <ic-button
+                class={{
+                  ["svg-container"]: true,
+                  ["dismiss-icon"]: true,
+                }}
+                innerHTML={dismissIcon}
+                onClick={this.dismissAction}
+                variant="icon"
+                size={size === "small" ? "small" : "default"}
+                appearance={IcThemeForegroundEnum.Dark}
+                title="dismiss"
```

Comment:
I tried using a screen reader on the clickable dismissible chip and when I focus on the whole chip, it reads out "Label dismiss". It might be reading out the tooltip text because the tooltip is a child of the chip `<button>` so wondering if a solution could be to move the close icon button outside and have it next to the `<Component>` rather than inside it (and then add styling so it appears inside it)? Not completely sure if that would work though but just an idea :)

Also just noticed that there are a couple of accessibility issues appearing in the Accessibility tab in storybook - one of which is about having nested focusable elements, so that solution might fix that issue